garak/resources/tap/generator_utils.py

```diff
@@ -51,22 +53,39 @@ def load_generator(
             generations=generations,
         )
     elif model_name in supported_huggingface:
-        generator = ConversationalPipeline(
-            model_name, generations=generations, device=device
-        )
+        generator = Model(model_name, generations=generations, device=device)
     else:
         msg = (
             f"{model_name} is not currently supported for TAP generation. Support is available for the following "
             f"OpenAI and HuggingFace models:\nOpenAI: {supported_openai}\nHuggingFace: {supported_huggingface}\n"
             f"Your jailbreaks will *NOT* be saved."
         )
         print(msg)
-        generator = ConversationalPipeline(
-            model_name, generations=generations, device=device
-        )
+        generator = Model(model_name, generations=generations, device=device)
 
     generator.max_tokens = max_tokens
     if temperature is not None:
         generator.temperature = temperature
 
     return generator
+
+
+def token_count(string: str, model_name: str) -> int:
+    encoding = tiktoken.encoding_for_model(model_name)
+    num_tokens = len(encoding.encode(string))
+    return num_tokens
+
+
+def get_token_limit(model_name: str) -> int:
```

Comment:
is this max generation length or max context length?

i think the latter.

if generation length, can it be extracted from the `max_len` param?


if context length.. would that this were available via openai API / generator (see #573 ). tagging that issue so maybe we pick up altering this code once context sizes are represented better (or at all) in garak.

Reply:
This is context length, yeah.